tests/test_salesforce_incremental_table_reset.py

```diff
@@ -0,0 +1,145 @@
+                replication_key = list(replication_keys[stream])[0]
+
+                # bookmarked states (top level objects)
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks').get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks').get(stream)
+
+                # bookmarked states (actual values)
+                first_bookmark_value = first_bookmark_key_value.get(replication_key)
+                second_bookmark_value = second_bookmark_key_value.get(replication_key)
+
+                # bookmarked values as utc for comparing against records
+                first_bookmark_value_utc = self.convert_state_to_utc(first_bookmark_value)
+                second_bookmark_value_utc = self.convert_state_to_utc(second_bookmark_value)
+
+                # Verify the second sync bookmark is Equal to the first sync bookmark
+                self.assertEqual(second_bookmark_value, first_bookmark_value) # assumes no changes to data during test
+
+                # Verify the second sync records respect the previous (simulated) bookmark value for the streams that are not reset
+                if stream != 'User' :
+                    simulated_bookmark_value = new_states['bookmarks'][stream][replication_key]
+                    for record in second_sync_messages:
+                        replication_key_value = record.get(replication_key)
+                        self.assertGreaterEqual(replication_key_value, simulated_bookmark_value,
+                                            msg="Second sync records do not repect the previous bookmark.")
+
+                # Verify the first sync bookmark value is the max replication key value for a given stream
+                for record in first_sync_messages:
+                    replication_key_value = record.get(replication_key)
+                    self.assertLessEqual(replication_key_value, first_bookmark_value_utc,
+                                         msg="First sync bookmark was set incorrectly, a record with a greater rep key value was synced")
+
+                # Verify the second sync bookmark value is the max replication key value for a given stream
+                for record in second_sync_messages:
+                    replication_key_value = record.get(replication_key)
+                    self.assertLessEqual(replication_key_value, second_bookmark_value_utc,
+                                         msg="Second sync bookmark was set incorrectly, a record with a greater rep key value was synced")
+
+                # Verify the number of records in the 2nd sync is equal to first for the stream that is reset and is less then the firsa for the restt
+                if stream != 'User':
+                    self.assertLess(second_sync_count, first_sync_count)
+                else:
+                    self.assertEqual(second_sync_count, first_sync_count)
```

Comment:
Same issue here with new data.

Reply:
For User stream, Changed the assertion to greaterEqual

Reply:
Added filtering to take care of record inserts between 2 syncs